### PR TITLE
TST: Add docstrings to arithmetic fixtures

### PR DESCRIPTION
**Pull request by dalgarno**
Relates to #19159

- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by datapythonista — COMMENTED** [review]
Thanks for the contribution @dalgarno.

This looks like an improvement, but after reading the docstrings it's still not clear what those fixture do, or how they should be used. Do you mind trying to rephrase, and may be use examples?

File: pandas/tests/arithmetic/conftest.py [hunk 1]
```diff
@@ -40,8 +57,19 @@ def one(request):
 
 @pytest.fixture(params=zeros)
 def zero(request):
-    # For testing division by (or of) zero for Index with length 5, this
-    # gives several scalar-zeros and length-5 vector-zeros
+    """
+    Several types of scalar zeros and length 5 vectors of zeros. For
+    testing division by (or of) zero.
+
+    Uses vector of length 5 for broadcasting with `numeric_idx` fixture,
```
Comment (WillAyd) [inline pandas/tests/arithmetic/conftest.py]: Can you model the extended description here after what you have in the other method? I find that quite nice, but don't really understand this one

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
lgtm. side issue we should rename these to maybe ``one_value``and so on and move to pandas/conftest.py (but for another PR)

**Comment by dalgarno** [comment]
@WillAyd can you approve if you're happy?

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @dalgarno

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @dalgarno

**Pull request merged**